ocsci/tests/test_config.py

```diff
@@ -12,8 +12,61 @@ def reset_config(self):
         ocsci.config.reset()
 
     def test_defaults(self):
-        ocsci.main.init_ocsci_conf()
         config_sections = ocsci.config.to_dict().keys()
         for section_name in config_sections:
             section = getattr(ocsci.config, section_name)
             assert section == getattr(ocs.defaults, section_name)
+
+    def test_custom_conf(self):
+        user_dict = dict(
+            REPORTING=dict(email='unit@test.com'),
+            RUN=dict(log_dir='/dev/null'),
+        )
+        ocsci.config.update(user_dict)
+        assert ocsci.config.REPORTING['email'] == 'unit@test.com'
+        assert ocsci.config.RUN['log_dir'] == '/dev/null'
+        assert ocsci.config.RUN['bin_dir'] == ocs.defaults.RUN['bin_dir']
```

Comment:
I want to avoid of duplications of sections like RUN, ENV_DATA, REPORTING in both places defaults.py module and the default_config.yaml file. I still don't see the reason why to keep updated on both places, so I would rather get rid of  default_config.yaml file and have all those default dictionaries (Sections) in defaults.py and then those should be used in the init of the config. We have then --ocsci-config and --cluster-config parameters to allow us overwrite those sections as well WDYT?

Reply:
I thought the plan was the have defaults in only one location?

Reply:
Yes, that what I told. All the defaults with all the sections like RUN and so on can live in defaults.py, you can use them in init of the config from defaults.py instead of from default_config.yaml as we do now. Then we can get rid of this file default_config.yaml and keep updating all the default values only in defaults.py moudule.

Reply:
ok, good - I see @vasukulkarni has a WIP in #218 for that.

Reply:
#218 is just moving this DEFAULTS section away from the default_config.yaml, but as we have now all the sections defined also in defaults.py I would get rid of this default_config.yaml and as I said, we can use sections from defaults.py for init the config.  Sounds ok @vasukulkarni @clacroix12 @zmc ?

Reply:
I don't see the point in keeping `defaults.py` either. We have our configuration object instantiated in `ocsci.__init__.py`. Why keep our default configuration values separate from where we instantiate the object?

Reply:
@petr-balogh oh, I thought he was going to drop the whole file in that PR.

@clacroix12 The defaults used by `Config` are coming from `defaults.py`

Reply:
@zmc once the things relevant to configuration are moved out I really just see a bunch of dictionaries that are loaded from yaml. I'm not really sure why these are in our `defaults` module in the first place. Maybe we can rename the module once the config stuff is out, or maybe there's a better home for these dictionaries. All I know is that the `defaults` module doesn't really have a clear purpose to me right now.

Reply:
we are going to keep defaults.py as it started with a purpose and it still provides a simple way to add things for qe folks, Also folks are writing tests using current model so we dont want to keep changing it behind the scenes.